pages/404.tsx

```diff
@@ -1,27 +1,30 @@
 import Link from "next/link";
 import { useEffect, useState } from "react";
-import { differenceInDays } from "date-fns";
 import FullHeightContainer from "components/atoms/FullHeightContainer/full-height-container";
 import HeaderLogo from "components/molecules/HeaderLogo/header-logo";
 import { useFetchTopContributors } from "lib/hooks/useFetchTopContributors";
 import DevCardWall from "components/organisms/DevCardWall/dev-card-wall";
-import { DevCardProps } from "components/molecules/DevCard/dev-card";
-import { fetchContributorPRs } from "lib/hooks/api/useContributorPullRequests";
-import { getRepoList } from "lib/hooks/useRepoList";
-import getContributorPullRequestVelocity from "lib/utils/get-contributor-pr-velocity";
-import getPercent from "lib/utils/get-percent";
-import { getAvatarByUsername } from "lib/utils/github";
 import BubbleBG from "../img/bubble-bg.svg";
+import { UserDevStats } from "./u/[username]/card";
+
+async function fetchUserData(username: string) {
+  const response = await fetch(`${process.env.NEXT_PUBLIC_API_URL}/users/${username}/devstats`, {
+    headers: {
+      "Content-Type": "application/json"
+    }
+  });
+  return await response.json() as UserDevStats;
+};
```

Comment:
Refactor (blocking): `fetchApiData` handles all of this. The username should also be validated before making the call.

You can use `fetchApiData`'s pathValidator

```javascript
pathValidator: () => safeParse(GitHubUserNameSchema, username).success
```

You can update validation-schemas.ts with this:

```diff
- import { parse, pipe, string, uuid } from "valibot";
+ import { parse, pipe, regex, string, uuid } from "valibot";

export const parseSchema = parse;

export const UuidSchema = pipe(string(), uuid("The UUID is badly formatted."));
+ 
+ const GitHubUserNameSchema = pipe(
+   string(),
+   regex(/^[a-z\d](?:[a-z\d]|-(?=[a-z\d])){0,38}$/i, "The GitHub + username is invalid.")
+ );
```